### ISSUE TITLE
Appium is returning wrong orientation for  Google Pixel C Tablet 

### ISSUE THREAD

**Issue by GrzegorzSidwa** [appium/appium#11663]
## The problem

There is a problem with screen orientation returned by appium on Google Pixel C tablet. When it's on landscape Appium is returning Portrait and the opposite. Issue exists on .Net and Java client. The issue is only on this specific device - it's not related to android version (already checked on another android 7 device). We created sample application which is returning current screen orientation to label text and it's working correctly, also on system rotation lock (notification bar on the top of the screen) we have information on current orientation and it's returned correctly. 
Can you please track how appium gets information about orientation and why Android returns it correctly (via app/notification bar) and Appium doesn't? 

Also screen rotation is working in opposite way. Trying to set landscape is rotating device to portrait and opposite.

Screen and logs attached:
![landscape orientation](https://user-images.githubusercontent.com/30998571/48263281-db5c9200-e425-11e8-9407-79d81035fbd0.png)

## Environment

* Appium version (or git revision) that exhibits the issue: 1.8.0, 1.9.2-beta.2
* Desktop OS/version used to run Appium:
* Node.js version (unless using Appium.app|exe): 7.7.3
* Mobile platform/version under test: Android 7 Google Pixel C tablet
* Real device or emulator/simulator: real device
* Appium CLI or Appium.app|exe: appium cli

## Link to Appium logs

https://gist.github.com/GrzegorzSidwa/da29c5593c35085f34f12bcdecb9980a

## Code To Reproduce Issue [ Good To Have ]

Please remember that with sample code it's easier to reproduce the bug and it's much faster to fix it.

Please git clone https://github.com/appium/appium and from the `sample-code` directory, use one of your favourite languages and sample apps to reproduce the issue.

In case a similar scenario is missing in sample-code, please submit a PR with one of the sample apps provided.

**Comment by mykola-mokhnach** [comment]
UIA@ driver retrieves current orientation via https://android.googlesource.com/platform/frameworks/base/+/master/core/java/android/view/Display.java#751 API and considers landscape mode if the returned value is either `ROTATION_90` or `ROTATION_270`:

https://github.com/appium/appium-uiautomator2-server/blob/d936c6de3ee93f8f7061e5098ce4d60f106e2219/app/src/main/java/io/appium/uiautomator2/handler/GetScreenOrientation.java#L20

Feel free to push a PR if you have ideas on how this algorithm can be improved.

**Comment by GrzegorzSidwa** [comment]
@mykola-mokhnach basically here situation is strange. Cause device is rotated, but orientation is returned as portrait.

**Comment by mykola-mokhnach** [comment]
Try to use rotation primitives in your client code instead of orientation as a workaround

**Comment by GrzegorzSidwa** [comment]
@mykola-mokhnach  not sure what do you mean by orientation primitives. My code looked like below:

`AppiumDriver<MobileElement> driver = new AppiumDriver<>(new URL("http://127.0.0.1:4723/wd/hub"), capabilities);
ScreenOrientation orientation = driver.getOrientation();
Dimension size = driver.manage().window().getSize();
driver.rotate(ScreenOrientation.PORTRAIT);
orientation = driver.getOrientation();
size = driver.manage().window().getSize();
`
In 2nd line Appium is returning orientation Portrait despite of Landscape which is recognized by Android System and our test app.

**Comment by mykola-mokhnach** [comment]
`size` might return a wrong value there. See https://github.com/appium/appium/issues/11656

**Comment by GrzegorzSidwa** [comment]
Yes, size is not an issue in this case. That's the reason why I've reported this issue.

**Comment by GrzegorzSidwa** [comment]
@mykola-mokhnach  did you have possibility of checking it? or maybe someone else?

**Comment by mykola-mokhnach** [comment]
I don't think that anybody from the team has time to check that.
But let me explain something first. Android OS itself operates the `rotation angle` entity and not an `orientation`. There are four angles: 0, 90, 180 and 270 degrees. We in appium assume that 90 and 270 are landscape orientations and 0, 180 are portrait, but this is not a hard restriction and device manufacturers may define their own criteria.

**Comment by mykola-mokhnach** [comment]
In appium@beta a new setting was added named `useResourcesForOrientationDetection`. If this setting is set to `true` then the server tries to use the Configuration object to determine device orientation rather than guess it based on the current rotation value. Please try it and let us know if that helps

**Comment by mykola-mokhnach** [comment]
Closed because of no response

**Comment by mirao** [comment]
I got the same issue on Pixel Tablet emulator with Android 11/14, in a CodeceptJS test, with the latest Appium/UIAutomator2.

I use the method https://codecept.io/helpers/Appium/#setorientation.
The call `I.setOrientation("PORTRAIT")` switches a view to "LANDSCAPE" and vice versa.
I tried `useResourcesForOrientationDetection: true`, but it causes that when the test starts my app, the view is automatically switched to PORTRAIT and it's not desired.

So I'm keeping the default config value (false) and in tests I'm using the "opposite" values for switching to appropriate modes. It works fine.
